src/js/components/OptionalSettingsComponent.jsx

```diff
@@ -0,0 +1,63 @@
+var React = require("react/addons");
+
+var appValidator = require("../validators/appValidator");
+var FormGroupComponent = require("../components/FormGroupComponent");
+
+var OptionalSettingsComponent = React.createClass({
+  displayName: "OptionalSettingsComponent",
+
+  propTypes: {
+    errors: React.PropTypes.array,
+    model: React.PropTypes.object.isRequired
+  },
+
+  render: function () {
+    let model = this.props.model;
+    let errors = this.props.errors;
+    let helpMessage = "Comma-separated list of valid constraints. Valid constraint format is \"field:operator[:value]\".";
```

Comment:
We should use `let` where it makes sense to have a variable declared within a scope ie:
```
for (let i = 0; i ...) {

}
```

Rather than every time we create a new variable. That way when we see it's very obvious.
What do you think?